core/cloudflow-operator/src/main/scala/cloudflow/operator/action/TopicActions.scala

```diff
@@ -56,69 +63,98 @@ object TopicActions {
         // find a config (secret) for the topic, all deployments should have the same configuration for the same topic,
         // since topics are defined once, deployments refer to the topics by port mappings.
         // So it is ok to just take the first one found, that uses the topic.
-        val secretName = newApp.spec.deployments
+        val appConfigSecretName = newApp.spec.deployments
           .filter(deploymentOf(topic.id))
           .headOption
           .map(_.secretName)
-        action(secretName, newApp.namespace, labels, topic)
+
+        action(appConfigSecretName, newApp.namespace, labels, topic)
       }
     actions
   }
 
   type TopicResource = ConfigMap
+  val DefaultConfigurationName = "default"
+  val KafkaClusterNameFormat   = "cloudflow-kafka-cluster-%s"
+  val KafkaClusterNameLabel    = "cloudflow.lightbend.com/kafka-cluster-name"
 
-  def action(secretName: Option[String], namespace: String, labels: CloudflowLabels, topic: TopicInfo)(
+  /**
+   * Create a topic using the correct Kafka configuration.
+   *
+   * Configuration resolution order:
+   * 1. App secret contains inline connection configuration for topic
+   * 2. User-defined topic cluster configuration name in blueprint
+   * 3. Default topic cluster configuration
+   */
+  def action(appConfigSecretName: Option[String], namespace: String, labels: CloudflowLabels, topic: TopicInfo)(
       implicit ctx: DeploymentContext
-  ): Action[ConfigMap] =
-    secretName
+  ): Action[ConfigMap] = {
+    def useClusterConfiguration(providedTopic: TopicInfo): Action[TopicResource] =
+      providedTopic.cluster
+        .map { cluster =>
+          Action.provided[Secret, ConfigMap](
+            String.format(KafkaClusterNameFormat, cluster),
+            ctx.podNamespace, {
+              case Some(secret) => createActionFromKafkaConfigSecret(secret, namespace, labels, providedTopic)
+              case None         =>
+                // TODO: cluster secret can't be found. we can either throw an exception and fail deployment (and maybe create a status message), or fallback to using the default Kafka configuration
+                throw new Exception(s"Could not find Kafka configuration for topic [${providedTopic.name}] cluster [$cluster]")
+            }
+          )
+        }
+        .getOrElse {
+          if (topic.cluster.eq(Some(DefaultConfigurationName)))
+            throw new Exception(
+              "A default Kafka configuration was not defined during installation of cloudflow-operator. Cannot create managed topics."
```

Comment:
It would be nice in the future to route all exceptions back to status updates in the operator, but that is definitely another issue/PR.